File: mac/Tests/DMShotTests/FramePresetsTests.swift
import XCTest
@testable import DMShot

final class FramePresetsTests: XCTestCase {
    func testPaddingFractions() {
        XCTAssertEqual(FramePresets.paddingFraction(.small), 0.04, accuracy: 1e-9)
        XCTAssertEqual(FramePresets.paddingFraction(.medium), 0.08, accuracy: 1e-9)
        XCTAssertEqual(FramePresets.paddingFraction(.large), 0.14, accuracy: 1e-9)
    }

    func testCornerFractions() {
        XCTAssertEqual(FramePresets.cornerFraction(.none), 0, accuracy: 1e-9)
        XCTAssertEqual(FramePresets.cornerFraction(.soft), 0.025, accuracy: 1e-9)
        XCTAssertEqual(FramePresets.cornerFraction(.round), 0.06, accuracy: 1e-9)
    }

    func testBlurConstants() {
        XCTAssertEqual(FramePresets.blurRadiusFraction, 0.06, accuracy: 1e-9)
        XCTAssertEqual(FramePresets.blurDarken, 0.12, accuracy: 1e-9)
    }

    func testSolidColors() {
        XCTAssertEqual(FramePresets.solidColors, ["#ffffff", "#ececec", "#2b2b2b", "#c97b4a"])
    }

    func testGradientStops() {
        XCTAssertEqual(FramePresets.gradientStops(.warm).0, "#f0883e")
        XCTAssertEqual(FramePresets.gradientStops(.warm).1, "#c0398a")
        XCTAssertEqual(FramePresets.gradientStops(.cool).0, "#3b82f6")
        XCTAssertEqual(FramePresets.gradientStops(.neutral).1, "#9a9a9a")
    }

    func testDefaultDisabledIsOff() {
        XCTAssertFalse(BackgroundStyle.disabled.enabled)
    }
}

File: windows/DMShot.Tests/FramePresetsTests.cs
using DMShot.Editor;
using Xunit;

public class FramePresetsTests
{
    [Fact]
    public void PaddingFractions()
    {
        Assert.Equal(0.04, FramePresets.PaddingFraction(FramePadding.Small), 9);
        Assert.Equal(0.08, FramePresets.PaddingFraction(FramePadding.Medium), 9);
        Assert.Equal(0.14, FramePresets.PaddingFraction(FramePadding.Large), 9);
    }

    [Fact]
    public void CornerFractions()
    {
        Assert.Equal(0.0, FramePresets.CornerFraction(FrameCorner.None), 9);
        Assert.Equal(0.025, FramePresets.CornerFraction(FrameCorner.Soft), 9);
        Assert.Equal(0.06, FramePresets.CornerFraction(FrameCorner.Round), 9);
    }

    [Fact]
    public void BlurConstants()
    {
        Assert.Equal(0.06, FramePresets.BlurRadiusFraction, 9);
        Assert.Equal(0.12, FramePresets.BlurDarken, 9);
    }

    [Fact]
    public void SolidColors()
    {
        Assert.Equal(new[] { "#ffffff", "#ececec", "#2b2b2b", "#c97b4a" }, FramePresets.SolidColors);
    }

    [Fact]
    public void GradientStops()
    {
        Assert.Equal("#f0883e", FramePresets.GradientStops(FrameGradient.Warm).Start);
        Assert.Equal("#c0398a", FramePresets.GradientStops(FrameGradient.Warm).End);
        Assert.Equal("#9a9a9a", FramePresets.GradientStops(FrameGradient.Neutral).End);
    }
}
